Compiled from "CWE396_Catch_Generic_Exception__Throwable_05.java"
public class testcases.CWE396_Catch_Generic_Exception.CWE396_Catch_Generic_Exception__Throwable_05 extends testcasesupport.AbstractTestCase {
  private boolean privateTrue;

  private boolean privateFalse;

  public testcases.CWE396_Catch_Generic_Exception.CWE396_Catch_Generic_Exception__Throwable_05();
    Code:
       0: aload_0
       1: invokespecial #1                  // Method testcasesupport/AbstractTestCase."<init>":()V
       4: aload_0
       5: iconst_1
       6: putfield      #7                  // Field privateTrue:Z
       9: aload_0
      10: iconst_0
      11: putfield      #13                 // Field privateFalse:Z
      14: return

  public void bad() throws java.lang.Throwable;
    Code:
       0: aload_0
       1: getfield      #7                  // Field privateTrue:Z
       4: ifeq          24
       7: ldc           #16                 // String Test
       9: invokestatic  #18                 // Method java/lang/Integer.parseInt:(Ljava/lang/String;)I
      12: pop
      13: goto          24
      16: astore_1
      17: ldc           #26                 // String Caught Throwable
      19: invokestatic  #28                 // Method testcasesupport/IO.writeLine:(Ljava/lang/String;)V
      22: aload_1
      23: athrow
      24: return
    Exception table:
       from    to  target type
           7    13    16   Class java/lang/Throwable

  private void good1() throws java.lang.Throwable;
    Code:
       0: aload_0
       1: getfield      #13                 // Field privateFalse:Z
       4: ifeq          15
       7: ldc           #34                 // String Benign, fixed string
       9: invokestatic  #28                 // Method testcasesupport/IO.writeLine:(Ljava/lang/String;)V
      12: goto          32
      15: ldc           #16                 // String Test
      17: invokestatic  #18                 // Method java/lang/Integer.parseInt:(Ljava/lang/String;)I
      20: pop
      21: goto          32
      24: astore_1
      25: ldc           #38                 // String Caught Exception
      27: invokestatic  #28                 // Method testcasesupport/IO.writeLine:(Ljava/lang/String;)V
      30: aload_1
      31: athrow
      32: return
    Exception table:
       from    to  target type
          15    21    24   Class java/lang/NumberFormatException

  private void good2() throws java.lang.Throwable;
    Code:
       0: aload_0
       1: getfield      #7                  // Field privateTrue:Z
       4: ifeq          24
       7: ldc           #16                 // String Test
       9: invokestatic  #18                 // Method java/lang/Integer.parseInt:(Ljava/lang/String;)I
      12: pop
      13: goto          24
      16: astore_1
      17: ldc           #38                 // String Caught Exception
      19: invokestatic  #28                 // Method testcasesupport/IO.writeLine:(Ljava/lang/String;)V
      22: aload_1
      23: athrow
      24: return
    Exception table:
       from    to  target type
           7    13    16   Class java/lang/NumberFormatException

  public void good() throws java.lang.Throwable;
    Code:
       0: aload_0
       1: invokevirtual #40                 // Method good1:()V
       4: aload_0
       5: invokevirtual #43                 // Method good2:()V
       8: return

  public static void main(java.lang.String[]) throws java.lang.ClassNotFoundException, java.lang.InstantiationException, java.lang.IllegalAccessException;
    Code:
       0: aload_0
       1: invokestatic  #46                 // Method mainFromParent:([Ljava/lang/String;)V
       4: return
}
